src/TimeWarpEventSet.cpp

```diff
@@ -84,9 +85,16 @@ void TimeWarpEventSet::insertEvent (unsigned int lp_id, std::shared_ptr<Event> e
 
         assert(input_queue_[lp_id]->size() == 1);
 
+
+#if defined (ONE_THREAD_PER_LTSF)
+#else
         schedule_queue_lock_[scheduler_id].lock();
+#endif
         schedule_queue_[scheduler_id]->insert(event);
+#if defined (ONE_THREAD_PER_LTSF)
+#else
         schedule_queue_lock_[scheduler_id].unlock();
+#endif
         scheduled_event_pointer_[lp_id] = event;
 
     } else {
```

Comment:
Inside the insertEvent() deactivate the entire else section i.e. the code chunk which looks at input queue to see whether there is any event available that is smaller than the scheduled event and tries to swap it. This call can be made by a thread other than the one linked to the schedule queue. So it will cause problems if there is no schedule queue lock.

Reply:
All whitespaces in your code should be spaces, it should not be tab. Please make sure of it. Editors suchas Visual Studio can be configured to place 4 space characters when the tab key is pressed.

Reply:
I made the changes inside insertEvent(). I'm pretty sure I got it correct, but it looks a bit confusing so I definetly need a check on that. Am currently changing all tabs to spaces